### DISCUSSION TITLE
Some QoL features

### DISCUSSION THREAD

**Discussion by CappiSteijns** [Zibbp/ganymede#108] [Ideas]
### The ability to fullscreen the VOD but still have the chat visible. 

When watching on a pc using firefox etc you can "F11" which makes it a lot better but when watching on my tablet (Samung S8 Ultra) it will look much more cleaner when I can fullscreen but also keep the chat visible.

### The ability to scroll back in the chat

Sometimes I want to scroll back in the chat but it instantly sets you back to the latest chat message. 

### The ability to edit a playlist

I can understand editing the `Name` of the playlist will mess up the database but what about the `Description`?
I found a typo but won't be able to make changes to the description.

### The ability to pause the vod on touching the screen on mobile.

In the browser you can quickly pause the VOD by pressing anywhere on the video. On mobile however you need to press the tiny little pause button on the navigation menu.

### Dark mode on default

Have the option to enable dark mode by default. A better welcome to new users visiting Ganymede.

### Confirmation message when clicking on "Save".

![image](https://user-images.githubusercontent.com/84481241/217259305-db93485a-c8a9-4d94-b973-673e62299f4b.png)

*please note that I am using Firefox on all devices, so maybe some of the idea's are already there but not when using Firefox.

I love this amazing project :D 

--- EDIT 01-03-2023:

### The ability to reorder the playlists.

![image](https://user-images.githubusercontent.com/84481241/222136088-6afb3cc9-a038-402e-b5cd-dc5e59c6e89b.png)

For example I would like to order the playlist with the years (2021, 2022, 2023 etc.) together.

### The ability to create non existing playlist when adding a VOD to a playlist that doesn't exist.

![image](https://user-images.githubusercontent.com/84481241/222138583-44562427-67a7-4d40-9196-f6a212192b4e.png)

This means you don't need to open a new tab to first create the playlist and go back to add the VOD to the playlist.

**Comment by Zibbp** [comment]
The mobile layout and navigation definitely needs some work. I mainly optimized for Desktop but I can take a look at improving the mobile functionality with your suggestions.

> The ability to scroll back in the chat

I can look into that functionality while the video is playing. For a workaround in the mean time you can pause the video and freely scroll. Right now there is a limit on the number of messages displayed in the chat window for performance but that can be changed.

> The ability to edit a playlist

This functionality already exists for the roles editor and admin I believe. When navigating to a playlist, there is a hamburger menu under the title which allows you to edit the name and description.
![brave_PVVZNSAd8U](https://user-images.githubusercontent.com/21207065/217279771-c5fb921a-dee9-45b3-a2ee-a7b498b0f3e4.png)

These are all good suggestions and I will probably implement them. I don't know when I'll get around to doing so.

**Reply by CappiSteijns** [reply]
Thank you for looking into it!

> This functionality already exists

Totally missed that, my mistake!

**Comment by Zibbp** [comment]
@CappiSteijns Better mobile support has been added in [v1.2.5](https://github.com/Zibbp/ganymede/releases/tag/v1.2.5). Let me know what you think and if you have any issues.

**Reply by CappiSteijns** [reply]
It's amazing. This will be way more comfortable watching on a bigger tablet!! Thank you for taking the time to do this, much appreciated

**Reply by CappiSteijns** [reply]
> It's amazing. This will be way more comfortable watching on a bigger tablet!! Thank you for taking the time to do this, much appreciated

So I had the chance to try it out on my Tablet. It doesn't really work great. It only works when I enable "Desktop site" in Firefox settings. When desktop enabled I get the full screen mode. But as you can see the buttons wont work anymore. 

I noticed the stream gets paused and unpaused on every single touch on the screen even if I make a swipe gesture. Can it be made it only pause/unpause when the user taps the vod?

Couldn't really figure out how to share the video I made to proof the point. So I share it using FireShare. 

https://lekkerspelenvods.cappiflex.nl/w/5e6cb8230882a9c61424b1a108a117a0

**Comment by Zibbp** [comment]
VideoJS does not have great touch UI built in and the plugin I found does not work with React. So I had to implement my own hacky touch UI.
> So I had the chance to try it out on my Tablet. It doesn't really work great. It only works when I enable "Desktop site" in Firefox settings. When desktop enabled I get the full screen mode. But as you can see the buttons wont work anymore.

I'm guessing this is due to my "smallDevice" check. Do you know the resolution of the device? I can increase this.
```javascript
const smallDevice = useMediaQuery("(max-width: 1000px)");
```

> I noticed the stream gets paused and unpaused on every single touch on the screen even if I make a swipe gesture. Can it be made it only pause/unpause when the user taps the vod?

I'm currently using a `onTouchStart` [event](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchstart_event) for playing/pausing the video. I can look into using `onTouchEnd` maybe that will help.

**Reply by CappiSteijns** [reply]
> I'm guessing this is due to my "smallDevice" check. Do you know the resolution of the device? I can increase this.

Resolution is `2960x1848` 

Looking forward using theater mode on my tablet ^^

**Reply by Zibbp** [reply]
[v1.2.6](https://github.com/Zibbp/ganymede/releases/tag/v1.2.6) has been released with an update that may allow you to "fullscreen theater" on your tablet. Not 100% sure it will work as I can't really test this thoroughly. As for the touch, I can't seem to find an API that will allow me to listen to normal touches, or rather I don't know how to go about doing that. I've never done any frontend for mobile development and VideoJS is a pain to work with especially with mobile support.
